woocommerce/woocommerce
# Subscription install improvements

### Changes proposed in this Pull Request:

This PR fixes a few edge case issues with subscription installations by catching errors and not interrupting ongoing installations. 

The install process on the frontend will now run until the subscription list is updated.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the [new subscriptions page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
2. Install a product
3. [Remove the product](http://localhost:8888/wp-admin/plugins.php)
4. Install it again 
5. All steps should work as expected



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Catch subscription installation errors and wait for the updated subscription list to load.

</details>
